src/consensus/aft/raft.h

```diff
@@ -1673,8 +1689,13 @@ namespace aft
       RequestVote rv{
         .term = state->current_view,
         .last_committable_idx = last_committable_idx,
-        .term_of_last_committable_idx = get_term_internal(last_committable_idx),
-        .election_type = election_type};
+        .term_of_last_committable_idx =
+          get_term_internal(last_committable_idx)};
+
+      if (election_type == ElectionType::PreVote)
+      {
+        rv.msg = RaftMsgType::raft_request_pre_vote;
```

Comment:
I think on-balance I'm fine with this approach, as it provides a small diff and simpler types, but it's worth noting that this is a little weird. The _type_ of `rv` is `RequestVote`, extending `RaftHeader<MsgType::request_vote>`. A different enum value appears in the type, and for every use _previously_ this value in the type implied the value in the instances. But we actually only use that to initialise the `msg` value, which is non-const, so we can override it here to get the behaviour we want. A little weird, a little spicy, but ultimately I don't see any imminent danger. This _could_ be changed to have a `const MsgType msg;` field, and then some other inheritance-or-nested-struct shenanigans to get distinct-types-with-matching-content for `RequestVote` and `RequestPreVote`.